app/services/store_sp_metadata_in_session.rb

```diff
@@ -5,6 +5,8 @@ def initialize(session:, request_id:)
   end
 
   def call
+    return unless sp_request
```

Comment:
Perhaps we can rename `EmailConfirmationTokenValidator` to `EmailConfirmationValidator` and add in a validation for the request ID.